build/check-images.ts

```diff
@@ -32,11 +34,11 @@ export function checkImageReferences(doc, $, options, { url, rawContent }) {
     // (addImageFlaw) is called two times. We can then assume the
     // findMatchesInText() will find it two times too. For each call,
     // we need to match the call based in counting matches from findMatchesInText().
-    const matches = [
-      ...findMatchesInText(src, rawContent, {
-        attribute: "src",
-      }),
-    ];
+    const matches = isMarkdown
+      ? findMatchesInMarkdown(rawContent, "image", src)
+      : findMatchesInText(src, rawContent, {
+          attribute: "src",
+        });
```

Comment:
As we still have many HTML tables in content. Is it necessary to match both in Markdown and Text for markdown files?

example: mdn/content#24154

Reply:
Hi @caugner, as we need to sort the matches, so the codes might be:

```ts
const matches = findMatchesInText(src, rawContent, {
  attribute: "src",
});
if (isMarkdown) {
  const hasMatchInHTML = matches.length > 0;
  matches.push(
    ...findMatchesInMarkdown(src, rawContent, { type: "image" })
  );
  // note that, we need to sort them by line and column
  if (hasMatchInHTML) {
    matches.sort((a, b) =>
      a.line === b.line ? a.column - b.column : a.line - b.line
    );
  }
}
```

I'm not sure could we call `findMatchesInText` function in `findMatchesInMarkdown` as we could deal with broken html format links in markdown file.

Reply:
Found that the [`fromMarkdown`](https://github.com/syntax-tree/mdast-util-from-markdown#syntax-tree) can also parse html nodes: see [AST](https://github.com/syntax-tree/mdast).

Resolved this by also parse HTML node in markdown